### PR TITLE
Scoped packages support

### PR DESCRIPTION
**Pull request by LINKIWI**
Update `npm` dependency to v2 and add support for shrinkwrapping scoped packages. New test added.

### CODE REVIEW

File: sync/purge-excess.js [hunk 1]
```diff
@@ -44,34 +45,69 @@ function findExcess(dir, shrinkwrap, opts, cb) {
...
+        return cb();
+    }
 
-            files = files.filter(function (file) {
-                // remove anything that is a dev dep
-                return devKeys.indexOf(file) === -1;
-            });
-        }
+    // consider top-level scoped packages only; e.g. those nested at the level
+    // node_modules/{*}
```
Comment (lxe) [inline sync/purge-excess.js]: why only top-level?

**Comment by lxe** [comment]
Did some smoke testing on large dependency trees involving git deps and namespace/scope prefixes. This works!

**Comment by LINKIWI** [comment]
Thanks @lxe! @Raynos can you review when you get a chance?

**Comment by lxe** [comment]
LGTM

**Comment by lxe** [comment]
Published as `npm-shrinkwrap@6.0.0`

**Pull request merged**